dill/_utils.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python
+#
+# Author: Leonardo Gama (@leogama)
+# Copyright (c) 2022 The Uncertainty Quantification Foundation.
+# License: 3-clause BSD.  The full license text is available at:
+#  - https://github.com/uqfoundation/dill/blob/master/LICENSE
+"""auxiliary internal classes used in multiple submodules, set here to avoid import recursion"""
+
+__all__ = ['AttrDict', 'ExcludeRules', 'Filter', 'RuleType']
+
+import logging
+logger = logging.getLogger('dill._utils')
+
+class AttrDict(dict):
+    """syntactic sugar for accessing dictionary items"""
+    def _check_attr(self, name):
```

Comment:
how important is the type checking?

Reply:
type checking? It checks if the the key isn't a built-in dict attribute before setting or deleting it. Will add a comment there.

Reply:
Yes, I should have said "attribute" checking...

Reply:
This class was removed.